pr-db-mongo/pr_db_mongo.py

```diff
@@ -0,0 +1,100 @@
+import subprocess
+import sys
+from urllib.parse import quote_plus, urlsplit, urlparse, urlunparse
+import os
+
+import pymongo
+
+PLUGIN_NAME = "pr-db-mongo"
+
+
+def execute_bash(command):
+    return subprocess.run(command, check=True, capture_output=True, text=True)
+
+def get_uri(db_url):
+    split_url = urlsplit(db_url)
+    username = split_url.username
+    password = split_url.password
+    scheme = split_url.scheme
+    hostname = split_url.hostname
+    port = split_url.port
+    query = split_url.query
+    credentials = ":".join([quote_plus(c) for c in (username, password)])
+    host = f"{hostname}:{port}" if port else hostname
+    options = f"/?{query}" if query else ""
+    return f"{scheme}://{credentials}@{host}{options}"
+
+def get_database_uri(mongo_uri, database_name):
+    parsed_uri = urlparse(mongo_uri)
+    uri_parts = list(parsed_uri)
+    uri_parts[2] = database_name
```

Comment:
1. Why mixing `urlparse`/`urlsplit`? `urlsplit` should be preferred (check the docs?
2. What is `uri_parts[2]`? Can't we use name for it?

Reply:
>  What is uri_parts[2]? Can't we use name for it?

@kilemensi the main aim of this function is to append database name to  given URI. I've tried to look into urlparse to find any method that could help in mutating the path property, but didn't find. A hack is to parse url, convert to list, update the third element and unparse the parts.

Reply:
My understanding @koechkevin the database you need to restore to isn't part of the URI (or doesn't need to be). You can use things like `--db` or `--nsInclude`/`--nsFrom` to specify it. URI is just to make sure we can connect. Isn't this correct?

Reply:
You are correct @kilemensi 
But we need config vars for the app as URI + database name

Reply:
[This](https://github.com/CodeForAfrica/dokku-plugins/blob/077d07574c4232b720af9328c0926fe920d473e7/pr-db-mongo/pr_db_mongo.py#L24) is where DB should go @koechkevin, no? i.e.

```js
    defaultauthdb = app_name
    options = f"?{query}" if query else ""
    return f"{scheme}://{credentials}@{host}/{defaultauthdb}{options}"
```

if `db` = `app_name`.

Reply:
Sometimes I need the URI without the database i.e with mongorestore/mongodump or else I it will end up with an error

Reply:
What error are you getting? [Docs](https://www.mongodb.com/docs/database-tools/mongorestore/#std-option-mongorestore.--uri) don't seem to indicate that specifying a DB in URI will result in error.

Reply:
Provide a different URI string and DB you get this error. So best way is to only provide only the URI.

![Uploading image.png…]()

Reply:
![image](https://github.com/CodeForAfrica/dokku-plugins/assets/39160236/68bfbb2f-b0e6-4df8-bad2-e40d676d29eb)

Reply:
Then pass `charter` as database when generating URI for `mongodump` and don't specify `--db`. What am I missing?